lib/project_types/theme/commands/common/root_helper.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Theme
+  class Command
+    module Common
+      module RootHelper
+        def root_value(args, options)
+          first = args&.first
+
+          if first.nil? || glob_params(options).include?(first)
+            "."
+          else
```

Comment:
@karreiro I hope you don't mind me jumping on here to review. 

This is an extremely small(!) edge case, but if somehow I typed
```
bundle exec shopify theme pull -d dist -x dist
```
then my root argument is ignored and a bunch of theme files are downloaded to my project root instead of my `dist` folder.

Generally, I feel like the root argument should take precedent and if any additional option conflicts with it or makes it ambiguous then there should be some kind of warning rather than having the root argument discarded.

With https://github.com/Shopify/shopify-cli/pull/2051, I'm really hoping to have my theme pull/push/serve all work from a `dist` folder like how Slate used to be setup, so the root argument is key there.

Otherwise this PR looks great and I definitely learned something new with the options converters. Thanks!

Reply:
@karreiro yes, as you suggested I actually meant...
```
shopify theme pull dist -d -x dist
```

The `-d` flag is fine to me as is (no need to support an additional flag param there)

( 
FWIW I think it's an artifact of the parser that because `-d` flag doesn't take a param the parser can still parse the next token that's not a flag as the first arg, so the command still works as intended either way

```
$ SHOPIFY_CLI_STACKTRACE=1 bundle exec shopify theme pull -d dist -x dist
⭑ You are running a development version of the CLI at:

pull...
args... ["dist"]
options... {:development=>true, :ignores=>["dist"]}
```

but I fully intend to stick with root option first as officially specified
)